pkg/services/sso/redhatsso_service.go

```diff
@@ -0,0 +1,136 @@
+	}
+	glog.V(5).Infof("Kafka Client %s deleted successfully", clientId)
+	return nil
+}
+
+func (r *redhatssoService) GetConfig() *keycloak.KeycloakConfig {
+	return nil
+}
+
+func (r *redhatssoService) GetRealmConfig() *keycloak.KeycloakRealmConfig {
+	return nil
+}
+
+func (r *redhatssoService) IsKafkaClientExist(accessToken string, clientId string) *errors.ServiceError {
+	return nil
+}
+
+func (r *redhatssoService) CreateServiceAccount(accessToken string, serviceAccountRequest *api.ServiceAccountRequest, ctx context.Context) (*api.ServiceAccount, *errors.ServiceError) {
+	serviceAccount, err := r.client.CreateServiceAccount(accessToken, serviceAccountRequest.Name, serviceAccountRequest.Description)
+	if err != nil {
+		return nil, errors.NewWithCause(errors.ErrorFailedToCreateServiceAccount, err, "failed to create service account")
+	}
+
+	return convertServiceAccountDataToAPIServiceAccount(&serviceAccount), nil
+}
+
+func (r *redhatssoService) DeleteServiceAccount(accessToken string, ctx context.Context, clientId string) *errors.ServiceError {
+	err := r.client.DeleteServiceAccount(accessToken, clientId)
+	if err != nil { //5xx
+		return errors.NewWithCause(errors.ErrorFailedToDeleteServiceAccount, err, "failed to delete service account")
+	}
+	glog.V(5).Infof("deleted service account clientId = %s ", clientId)
+	return nil
+}
+
+func (r *redhatssoService) ResetServiceAccountCredentials(accessToken string, ctx context.Context, clientId string) (*api.ServiceAccount, *errors.ServiceError) {
+	serviceAccount, err := r.client.RegenerateClientSecret(accessToken, clientId)
+	if err != nil { //5xx
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to reset service account credentials")
+	}
+	return convertServiceAccountDataToAPIServiceAccount(&serviceAccount), nil
+}
+
+func (r *redhatssoService) ListServiceAcc(accessToken string, ctx context.Context, first int, max int) ([]api.ServiceAccount, *errors.ServiceError) {
+	serviceAccounts, err := r.client.GetServiceAccounts(accessToken, first, max)
+	if err != nil {
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to collect service accounts")
+	}
+
+	var saList []api.ServiceAccount
+	for _, sa := range serviceAccounts {
+		saList = append(saList, *convertServiceAccountDataToAPIServiceAccount(&sa))
+	}
+	return saList, nil
+}
+func (r *redhatssoService) RegisterKasFleetshardOperatorServiceAccount(accessToken string, agentClusterId string) (*api.ServiceAccount, *errors.ServiceError) {
+	return nil, nil
+}
+func (r *redhatssoService) DeRegisterKasFleetshardOperatorServiceAccount(accessToken string, agentClusterId string) *errors.ServiceError {
+	return nil
+}
+func (r *redhatssoService) GetServiceAccountById(accessToken string, ctx context.Context, id string) (*api.ServiceAccount, *errors.ServiceError) {
+	return r.GetServiceAccountByClientId(accessToken, ctx, id)
+}
```

Comment:
@akoserwal Is it ok to use the `id` as `clientId`? Or I should loop through all the service accounts and search for the one that has `sa,Id == id` ?

Reply:
yes